Tests/ML/test_lightning_containers.py

```diff
@@ -312,3 +313,36 @@ def test_get_hyperdrive_config(number_of_cross_validation_splits: int,
     else:
         hd_config = container.get_hyperdrive_config(run_config=run_config)
         assert isinstance(hd_config, HyperDriveConfig)
+
+
+@pytest.mark.parametrize("allow_partial_ground_truth", [True, False])
+def test_innereyecontainer_setup_passes_on_allow_incomplete_labels(
+        test_output_dirs: OutputFolderForTests,
+        allow_partial_ground_truth: bool) -> None:
+    """
+    Test that InnerEyeContainer.setup passes on the correct value of allow_incomplete_labels to
+    full_image_dataset.convert_channels_to_file_paths
+    :param test_output_dirs: Test fixture.
+    :param allow_partial_ground_truth: The value to set allow_incomplete_labels to and check it is
+    passed through.
+    """
+    config = DummyModel()
+    config.set_output_to(test_output_dirs.root_dir)
+    config.allow_incomplete_labels = allow_partial_ground_truth
+    container = InnerEyeContainer(config)
+    test_done_message = "Stop now, the test has passed."
+
+    def mocked_convert_channels_to_file_paths(
+            _: List[str],
+            __: pd.DataFrame,
+            ___: Path,
+            ____: str,
+            allow_incomplete_labels: bool) -> Tuple[List[Optional[Path]], str]:
+        assert allow_incomplete_labels == allow_partial_ground_truth
+        raise RuntimeError(test_done_message)
+
+    with pytest.raises(RuntimeError) as runtime_error:
+        with mock.patch("InnerEye.ML.lightning_base.convert_channels_to_file_paths") as convert_channels_to_file_paths_mock:
+            convert_channels_to_file_paths_mock.side_effect = mocked_convert_channels_to_file_paths
+            container.setup()
+    assert str(runtime_error.value) == test_done_message
```

Comment:
Rather than raising an exception, is it possible to use any of the mock built-in functions like `assert_called`? https://docs.python.org/3/library/unittest.mock.html

Reply:
No, I need to stop the container.setup code, rather than needing to mock up a reasonable return value from mocked_convert_channels_to_file_paths